utility/utils.py

```diff
@@ -50,3 +52,27 @@ def get_config_root_full_path():
 
     config_root = config_root.split("~/")[1]
     return f"{get_noobaa_sa_host_home_path()}/{config_root}"
+
+
+def check_data_integrity(origin_dir, results_dir):
+    """
+    Ckeck the data integrity of downloaded objects with uploaded objects
+    """
+
+    uploaded_objs_names = os.listdir(origin_dir)
+    downloaded_objs_names = os.listdir(results_dir)
+
+    assert (
+        len(uploaded_objs_names) == len(downloaded_objs_names)
+    ), "Downloaded and original objects count does not match"
+    uploaded_objs_names.sort()
+    downloaded_objs_names.sort()
+    for uploaded, downloaded in zip(
+        uploaded_objs_names, downloaded_objs_names
+    ):
+        original_full_path = os.path.join(origin_dir, uploaded)
+        downloaded_full_path = os.path.join(results_dir, downloaded)
+        assert (
+            compare_md5sums(original_full_path, downloaded_full_path)
+        ), f"MD5 sums do not match for {uploaded}"
+        log.info(f"MD5sums are matched for object {uploaded} and {downloaded}")
```

Comment:
Looks good, but let's return a boolean instead of making the asserts in this function itself, and change the name and Docstring of the func accordingly.

Besides allowing using the logic here with more versatility, it helps establishing a more correct design where we make asserts only in the tests themselves. Besides "correctness", it will help keeping util funcs like this versatile by default.

Reply:
Fixed